amaranth/_utils.py

```diff
@@ -74,18 +74,39 @@ def decorator_like(*args, **kwargs):
                 return f(*args, **kwargs)
         return decorator_like
 
+_LINT_PATTERN = re.compile(r"^#\s*amaranth:\s*((?:\w+=\w+\s*)(?:,\s*\w+=\w+\s*)*)\n$")
+
 
 def get_linter_options(filename):
     first_line = linecache.getline(filename, 1)
     if first_line:
-        match = re.match(r"^#\s*amaranth:\s*((?:\w+=\w+\s*)(?:,\s*\w+=\w+\s*)*)\n$", first_line)
+        match = _LINT_PATTERN.match(first_line)
         if match:
             return dict(map(lambda s: s.strip().split("=", 2), match.group(1).split(",")))
     return dict()
 
 
-def get_linter_option(filename, name, type, default):
-    options = get_linter_options(filename)
+def get_linter_option(frame, name, type, default):
+    """
+    Get given linter option for a given stack frame. This iterates down the frames, collating options of the form:
+    ```
+    # amaranth: {name} = value
```

Comment:
This isn't the right syntax: we don't allow spaces around `=`.

Reply:
fixed